In [6]:
import pandas as pd # import pandas library for data manipulation and analysis
from pathlib import Path

# Use Path.cwd().parent to go up 1 folder level
BASE_DIR = Path.cwd().parent
DATA_PATH = BASE_DIR / "data" / "WA_Fn-UseC_-Telco-Customer-Churn.csv"

#print(BASE_DIR)
Telco_cust_churn_data = pd.read_csv(DATA_PATH) # read the CSV file into a pandas DataFrame




In [7]:
Telco_cust_churn_data.head() # display the first few rows of the DataFrame

,customerID,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,...,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,MonthlyCharges,TotalCharges,Churn
0,7590-VHVEG,Female,0,Yes,No,1,No,No phone service,DSL,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,29.85,29.85,No
1,5575-GNVDE,Male,0,No,No,34,Yes,No,DSL,Yes,...,Yes,No,No,No,One year,No,Mailed check,56.95,1889.5,No
2,3668-QPYBK,Male,0,No,No,2,Yes,No,DSL,Yes,...,No,No,No,No,Month-to-month,Yes,Mailed check,53.85,108.15,Yes
3,7795-CFOCW,Male,0,No,No,45,No,No phone service,DSL,Yes,...,Yes,Yes,No,No,One year,No,Bank transfer (automatic),42.30,1840.75,No
4,9237-HQITU,Female,0,No,No,2,Yes,No,Fiber optic,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,70.70,151.65,Yes


In [8]:
Telco_cust_churn_data.info() # display information about the DataFrame, including column names, data types, and non-null counts

<class 'pandas.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7043 non-null   str    
 1   gender            7043 non-null   str    
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   str    
 4   Dependents        7043 non-null   str    
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   str    
 7   MultipleLines     7043 non-null   str    
 8   InternetService   7043 non-null   str    
 9   OnlineSecurity    7043 non-null   str    
 10  OnlineBackup      7043 non-null   str    
 11  DeviceProtection  7043 non-null   str    
 12  TechSupport       7043 non-null   str    
 13  StreamingTV       7043 non-null   str    
 14  StreamingMovies   7043 non-null   str    
 15  Contract          7043 non-null   str    
 16  PaperlessBilling  7043 non-null   str    
 17  Paymen

In [9]:
Telco_cust_churn_data.shape # display the shape of the DataFrame (number of rows and columns)

(7043, 21)

In [10]:
Telco_cust_churn_data.dtypes

customerID              str
gender                  str
SeniorCitizen         int64
Partner                 str
Dependents              str
tenure                int64
PhoneService            str
MultipleLines           str
InternetService         str
OnlineSecurity          str
OnlineBackup            str
DeviceProtection        str
TechSupport             str
StreamingTV             str
StreamingMovies         str
Contract                str
PaperlessBilling        str
PaymentMethod           str
MonthlyCharges      float64
TotalCharges            str
Churn                   str
dtype: object

In [11]:
Telco_cust_churn_data.describe() # display summary statistics for numerical columns in the DataFrame

,SeniorCitizen,tenure,MonthlyCharges
count,7043.000000,7043.000000,7043.000000
mean,0.162147,32.371149,64.761692
std,0.368612,24.559481,30.090047
min,0.000000,0.000000,18.250000
25%,0.000000,9.000000,35.500000
50%,0.000000,29.000000,70.350000
75%,0.000000,55.000000,89.850000
max,1.000000,72.000000,118.750000


# summary statistics for numerical columns for total 7043 observation
## SeniorCitizen
### 1] 16% observation are senior citizen
### 2] as it is binary value only 0/1 is min/max

## tenure
### 1] customer stays witth company is 32.37 ~2.7 year
### 2] min is 0 means some customer are new
### 3] max is 72 , some customer stays for long duration ~ 6 year
### 4] 9 , 29, 55 --> 25 percentile are for < 1 year while 50 percentile stay with < 2 year 5 month even 75% intouch < 4 year 7 month

## MonthlyCharges
### 1] avg unit amount is 64.76
### 2] min / max is 18.25 / 118.750000 unit charges per month
### 3] 25%, 50%, 70% --> 35, 70, 89 pay for month

## as std value
### 1] SeniorCitizen has 0.3686 low spread near 0
### 2] tenure 24.56 month < mean 32.37 , diverse customer lifespans
### 3] MonthlyCharges has 30 big variation in billing amounts as far from mean 64.76